hack/e2e/run.sh

```diff
@@ -0,0 +1,150 @@
+# Otherwise, rely on KUBECONFIG to determine test cluster.
+UP="${UP:-yes}"
+# if DOWN==yes, delete cluster after test
+DOWN="${DOWN:-yes}"
+
+KUBERNETES_VERSION="${KUBERNETES_VERSION:-v1.23.2}"
+GINKGO_VERSION="v1.14.0"
+CLUSTER_NAME="test-cluster-${test_run_id}.k8s.local"
+KOPS_STATE_STORE="${KOPS_STATE_STORE:-}"
+REGION="${AWS_REGION:-us-west-2}"
+ZONES="${AWS_AVAILABILITY_ZONES:-us-west-2a,us-west-2b,us-west-2c}"
+AWS_ACCOUNT_ID=$(aws sts get-caller-identity --query Account --output text)
+IMAGE_NAME=${IMAGE_NAME:-${AWS_ACCOUNT_ID}.dkr.ecr.${REGION}.amazonaws.com/amazon/cloud-controller-manager}
+# $VERSION is set in Makefile
+IMAGE_TAG=${IMAGE_TAG:-${VERSION}-${test_run_id}}
+
+# Test args
+GINKGO_FOCUS=${GINKGO_FOCUS:-"\[cloud-provider-aws-e2e\]"}
+GINKGO_SKIP=${GINKGO_SKIP:-"\[Disruptive\]"}
+GINKGO_NODES=${GINKGO_NODES:-4}
+
+EXPANDED_TEST_EXTRA_FLAGS="${EXPANDED_TEST_EXTRA_FLAGS:-}"
+
+if [[ -z "${KOPS_STATE_STORE}" ]]; then
+    echo "KOPS_STATE_STORE must be set"
+    exit 1
+fi
+
+if [[ ! -f "${repo_root}/e2e.test" ]]; then
+    echo "Missing e2e.test binary"
+    exit 1
+fi
+
+if [[ -z "${SSH_PUBLIC_KEY_PATH}" ]]; then
+    ssh_key_path=${test_run}/sshkey
+    ssh-keygen -b 2048 -t rsa -f ${ssh_key_path} -q -N ""
+    SSH_PUBLIC_KEY_PATH=${ssh_key_path}.pub
+fi
+
+if [[ "${UP}" =~ "^(yes|no)$" ]]; then
+    echo "Creating cluster: ${UP}"
+else
+    echo "Invalid UP: ${UP} (valid: [yes|no])"
+fi
+
+if [[ "${DOWN}" =~ "^(yes|no)$" ]]; then
+    echo "Deleting cluster: ${DOWN}"
+else
+    echo "Invalid DOWN: ${DOWN} (valid: [yes|no])"
+fi
+
+echo "Starting test run ---"
+echo " + Region:              ${REGION} (${ZONES})"
+echo " + Cluster name:        ${CLUSTER_NAME}"
+echo " + Kubernetes version:  ${KUBERNETES_VERSION}"
+echo " + Focus:               ${GINKGO_FOCUS}"
+echo " + Skip:                ${GINKGO_SKIP}"
+echo " + Kops state store:    ${KOPS_STATE_STORE}"
+echo " + SSH public key path: ${SSH_PUBLIC_KEY_PATH}"
+echo " + Test run ID:         ${test_run_id}"
+echo " + Kubetest run dir:    ${test_run}"
+echo " + Image:               ${IMAGE_NAME}:${IMAGE_TAG}"
+echo " + Up:                  ${UP}"
+echo " + Down:                ${DOWN}"
+
+mkdir -p "${test_run}"
+
+export KOPS_STATE_STORE
+# kubetest2 sets RunDir as filepath.Join(artifacts.BaseDir(), o.RunID())
+export ARTIFACTS="${test_output_root}"
+export KUBETEST2_RUN_DIR="${test_run}"
+export PATH="${PATH}"
+
+echo "Installing e2e.test to ${test_run}"
+cp "${repo_root}/e2e.test" "${test_run}"
+
+echo "Installing ginkgo to ${test_run}"
+GINKGO_BIN="${test_run}/ginkgo"
+if [[ ! -f ${GINKGO_BIN} ]]; then
+  GOBIN=${test_run} go install "github.com/onsi/ginkgo/ginkgo@${GINKGO_VERSION}"
+fi
+
+echo "Building and pushing test driver image to ${IMAGE_NAME}:${IMAGE_TAG}"
+aws ecr get-login-password --region "${REGION}" | docker login --username AWS --password-stdin "${AWS_ACCOUNT_ID}.dkr.ecr.${REGION}.amazonaws.com"
+docker tag "${IMAGE}" "${IMAGE_NAME}:${IMAGE_TAG}"
```

Comment:
is IMAGE expected to be an arg? no magic allowed, it should be defined at the top with other args

Reply:
might need some prefixes to disambiguate IMAGE from IMAGE_NAME and IMAGE_TAG... IMAGE is the image that is assumed t o exist, IMAGE_NAME:IMAGE_TAG is the destination to push it to

Reply:
Updated to use `$MAKE_` prefix to differentiate variables passed from Makefile.